src/datatypes.ts

```diff
@@ -153,6 +156,110 @@ export function deserializeDatetime(literalString: string): Date | null {
   return date;
 }
 
+/**
+ * Time type for time data type attributes
+ */
+export type Time = {
+  hour: number;
+  minute: number;
+  second: number;
+  millisecond?: number;
+  timezoneHourOffset?: number;
+  timezoneMinuteOffset?: number;
+};
+
+/**
+ * @internal
+ * @param value Value to serialise.
+ * @returns String representation of `value` in UTC.
+ * @see https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
+ */
+export function serializeTime(value: Time): string {
+  const utcHours =
+    value.hour + (value.timezoneHourOffset ? value.timezoneHourOffset : 0);
+  const utcMinutes =
+    value.minute +
+    (value.timezoneMinuteOffset ? value.timezoneMinuteOffset : 0);
+
+  let millisecondString;
+  // get millisecond into three digits
+  if (value.millisecond) {
+    if (value.millisecond < 10) {
+      millisecondString = "00" + value.millisecond;
+    } else if (value.millisecond < 100) {
+      millisecondString = "0" + value.millisecond;
+    } else {
+      millisecondString = value.millisecond;
+    }
+  }
+
+  return (
+    (utcHours < 10 ? "0" + utcHours : utcHours) +
+    ":" +
+    (utcMinutes < 10 ? "0" + utcMinutes : utcMinutes) +
+    ":" +
+    (value.second < 10 ? "0" + value.second : value.second) +
+    (value.millisecond ? "." + millisecondString + "Z" : "Z")
+  );
+}
+
+/**
+ * @internal
+ * @param literalString Value to deserialise.
+ * @returns Deserialized time, or null if the given value is not a valid serialised datetime.
+ * @see https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
+ */
+export function deserializeTime(literalString: string): Time | null {
+  // Time in the format described at
+  // https://www.w3.org/TR/xmlschema-2/#time-lexical-representation
+  // (without constraints on the value).
+  // \d\d:\d\d:\d\d - Two digits for the hour, minute and second, respectively, separated by a `:`.
+  //                  Example: "13:37:42".
+  // (\.\d+)? - Optionally a `.` followed by one or more digits representing milliseconds.
+  //            Example: ".1337".
+  // (Z|(\+|-)\d\d:\d\d) - The letter Z indicating UTC, or a `+` or `-` followed by two digits for
+  //                       the hour offset and two for the minute offset, separated by a `:`.
+  //                       Example: "+13:37".
+  const datetimeRegEx = /\d\d:\d\d:\d\d(\.\d+)?(Z|(\+|-)\d\d:\d\d)/;
+  if (!datetimeRegEx.test(literalString)) {
+    return null;
+  }
+  const [timeString, timezoneString] = splitTimeFromTimezone(literalString);
+  const [hourOffset, minuteOffset] = getTimezoneOffsets(timezoneString);
+  if (minuteOffset > 59) {
+    return null;
+  }
+  const [hourString, minuteString, timeRest] = timeString.split(":");
+  let utcHours = Number.parseInt(hourString, 10) + hourOffset;
+  let utcMinutes = Number.parseInt(minuteString, 10) + minuteOffset;
```

Comment:
Here too I think it's probably better to return the original hour and minute values, and the timezone offsets separately, to avoid losing data.

Reply:
Okay, will do both. I only chose to do it this way because we store datetime and date as UTC using this method. If you think that this isn't the best solution for time then I'll get it changed.

Reply:
Yeah as mentioned elsewhere, when using `Date` the developer can't explicitly choose a time zone, so there's no information we're throwing away there. Once [Temporal](https://github.com/tc39/proposal-temporal) makes it to JavaScript and we add support, we'll have to preserve the timezone too for Date and Datetime when passed it.